behaviour/graql/language/define.feature

```diff
@@ -1018,92 +928,700 @@ Feature: Graql Define Query
+  Scenario: define a rule with nested negation throws on commit
+    Given graql define
+      """
+      define
+      nickname sub name;
+      person has nickname;
+      """
+    Given the integrity is validated
+    Then graql define throws
+      """
+      define
+      robert-doesnt-not-have-nickname-bob sub rule,
+      when {
+        not {
+          not {
+            $p has name "Robert";
+          };
+        };
+      }, then {
+        $p has nickname "Bob";
+      };
+      """
 
-  #########################
-  # TODO: SCHEMA MUTATION #
-  #########################
 
+  Scenario: define a rule with two conclusions throws on commit
+    Given graql define
+      """
+      define
+      nickname sub name;
+      person has nickname;
+      """
+    Given the integrity is validated
+    Then graql define throws
+      """
+      define
+      robert-has-nicknames-bob-and-bobby sub rule,
+      when {
+        $p has name "Robert";
+      }, then {
+        $p has nickname "Bob";
+        $p has nickname "Bobby";
+      };
+      """
+
+
+  Scenario: define a rule with conjunction creates a rule
+    Given graql define
+      """
+      define
+      person plays named-robert;
+      both-named-robert sub relation, relates named-robert;
+      two-roberts-are-both-named-robert sub rule,
+      when {
+        $p isa person, has name "Robert";
+        $p2 isa person, has name "Robert";
+      }, then {
+        (named-robert: $p, named-robert: $p2) isa both-named-robert;
+      };
+      """
+    Given the integrity is validated
+    When get answers of graql query
+      """
+      match $x sub rule; get;
+      """
+    Then concept identifiers are
+      |     | check | value                             |
+      | BOB | label | two-roberts-are-both-named-robert |
+      | RUL | label | rule                              |
+    Then uniquely identify answer concepts
+      | x   |
+      | BOB |
+      | RUL |
+
+
+  Scenario: define a rule with disjunction throws on commit
+    Given graql define
+      """
+      define
+      nickname sub name;
+      person has nickname;
+      """
+    Given the integrity is validated
+    Then graql define throws
+      """
+      define
+      sophie-and-fiona-have-nickname-fi sub rule,
+      when {
+        {$p has name "Sophie";} or {$p has name "Fiona";};
+      }, then {
+        $p has nickname "Fi";
+      };
+      """
+
+
+  Scenario: define rule with an unbound variable in the `then` throws on commit
+    Given graql define
+      """
+      define
+      nickname sub name;
+      person has nickname;
+      """
+    Given the integrity is validated
+    Then graql define throws
+      """
+      define
+      i-did-a-bad-typo sub rule,
+      when {
+        $p has name "I am a person";
+      }, then {
+        $q has nickname "Who am I?";
+      };
+      """
+
+
+  Scenario: define rule with an undefined attribute set in `then` throws on commit
```

Comment:
waaaait this throws??

Reply:
Yes. It should, shouldn't it? `person` doesn't have an attribute called `age` so it's an invalid rule

Reply:
ah ok, cool